docs/src/darray.md

```diff
@@ -0,0 +1,135 @@
+# Distributed Array
+
+The `DArray`, or "distributed table", is an abstraction layer on top of Dagger
+that allows loading table-like structures into a distributed environment. 
+
+This should not be confused with the [DistributedArrays.jl](https://github.com/JuliaParallel/DistributedArrays.jl) package.
+
+## Creating a `DArray`
+
+Distributed arrays are created by invoking `compute` on an operation resulting in a distributed array.
+
+For example, one can allocate an array in a distributed fashion in the following fashion. To obtain a normal `Array`,
+`collect` can be used to gather the data from the various processes.
+```julia
+julia> DX = rand(Blocks(50,50), 100, 100)
+Dagger.AllocateArray{Float64, 2}(100, 100)
+
+julia> compute(DX)
+Dagger.DArray{Any, 2, typeof(cat)}(100, 100)
```

Comment:
Following from the above comment, maybe we then say, "This call causes Dagger to actually allocate the randomly-initialized `DArray`. Without this call, the `DArray` is not yet materialized.".